src/containers/upload-page/store/sagas.js

```diff
@@ -488,6 +494,21 @@ export function* handleUploads(tracks, isCollection, isStem = false) {
   yield all(workerTasks.map(t => cancel(t)))
 
   let returnVal = { trackIds }
+
+  // Report success + failure events
+  const uploadType = isCollection
+    ? isAlbum
+      ? 'album'
+      : 'playlist'
+    : 'multi_track'
+  yield reportSuccessAndFailureEvents({
+    // Don't report non-uploaded tracks due to playlist upload abort
+    numSuccess: tracks.length - failedRequests.length - numOutstandingRequests,
```

Comment:
seems a little dangerous to do this math rather than tracking a number through. i could see us intro-ing a bug in the tracking of failures or something and then losing successes. i can't think of anything better though.

Reply:
Yeah, this worried me a little bit too. It's definitely fine as it is now but I'm always down w defensive programming; it'd be easy to just add a count here. I'll do it